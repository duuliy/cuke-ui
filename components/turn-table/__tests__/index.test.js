import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import TurnTable from "../index";

const prizes = new Array(8).fill(0).map((_, i) => `奖品${i + 1}`);

describe("<TurnTable/>", () => {
  it("should render a <TurnTable/> components", () => {
    const wrapper = render(<TurnTable prizes={prizes} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-turn-table classnames", () => {
    const wrapper = shallow(<TurnTable prizes={prizes} />);
    assert(wrapper.find(".cuke-turn-table").length === 1);
  });

  it("should render the custom config", () => {
    const wrapper = render(
      <TurnTable prizes={prizes} clickText="我要抽奖" prefixCls="custom" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
