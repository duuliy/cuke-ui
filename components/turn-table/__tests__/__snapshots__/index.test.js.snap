// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TurnTable/> should render a <TurnTable/> components 1`] = `
<div
  class="react-turntable react-turntable-section cuke-turn-table"
  style="width:500px;height:500px"
>
  <canvas
    id="react-turntable-section-canvas"
  />
  <div
    class="react-turntable-section-btn"
  >
    抽奖
  </div>
</div>
`;

exports[`<TurnTable/> should render the custom config 1`] = `
<div
  class="react-turntable react-turntable-section custom"
  style="width:500px;height:500px"
>
  <canvas
    id="react-turntable-section-canvas"
  />
  <div
    class="react-turntable-section-btn"
  >
    我要抽奖
  </div>
</div>
`;
